src/ivc/public_params.rs

```diff
@@ -58,21 +58,18 @@ where
     fn new(
         k_table_size: u32,
         commitment_key: &'key CommitmentKey<C>,
-        is_primary_circuit: bool,
```

Comment:
In this case, we will not return default `U_new_base` for the secondary but create `U_new_base` from `inputs.u` same as primary?

Reply:
We now have to pass `primary_u` immediately as a relax circuit to the input of the zero round of secondary circuit

This API call looks different, but in fact we did the same thing earlier, stretching this transformation over several abstraction layers

Thus `U` to be passed to the input will always be returned from base-case

Reply:
I see. This makes the circuit easier. The tradeoff is that we have one more step because at the base case, we don't do the folding of input `u` with default `U`.

Not sure if it worth the effort to save one step, we can add a comment here for potential future optimization.

Reply:
No, the only compromise inside is that `U` is now passed outside, but folding we do the same thing. 

I mean, the folding has always been the same.

Reply:
That is, previously it took the `U: default` version OR the incoming `u.to_relax()` version and ignored the incoming `U` version

Now it always takes the incoming `U`